kwanj-k/storemanager-v2
# #161404710 Add authentication endpoints

### What does this PR do?
Add all the authentication related endpoints

#### Description of Task to be Completed?
-- Create a registration/signup endpoint
-- Create a login endpoint
-- Create endpoints for adding an admin/attendant

### How should this be manually tested?
* After cloning the repo, git pull the branch ft-auth-#161404710 
* Install a virtual enviroment and activate it [Refer here](https://docs.python-guide.org/dev/virtualenvs/)
* Pip install the the requirements.txt
* Run the following:
  ``` export FLASK_APP='run.py' ```
  ``` export APP_SETTINGS='development' ```
   ``` pytest --cov-report term-missing --cov=app ```

Also, you need to create a development and test database and connect to it.[Refer here](https://www.digitalocean.com/community/tutorials/how-to-install-and-use-postgresql-on-ubuntu-18-04#creating-a-new-role)

Then export the database URL in the environment:
```
export Dev_URL="dbname=dbname user=username password=password"
export Test_URL="dbname=dbname user=username password=password"
export SECRET="iamasecrettheunsadtheunsulliedtheundead"

```
We need the secret key for generating the tokens

Then Run the tests:
```
pytest --cov-report term-missing --cov=app
```
Four tests should now pass as shown:
![screenshot from 2018-10-24 20-21-06](https://user-images.githubusercontent.com/33450849/47449070-643ec100-d7ca-11e8-936f-8e9681d60e32.png)


### What are the relevant Pivotal Tracker stories?
  #161404710
